File: src/main/java/club/deneb/client/irc/IrcManager.java
package club.deneb.client.irc;

import club.deneb.client.features.modules.client.IRC;

public class IrcManager {

    public IrcClient client;

    public final String proxy = "irc.deneb.club";
    public final int port = 20730;

    public IrcManager(){
        client = new IrcClient(proxy,port);
        client.start();
    }

    public void reconnect(){
        if(client != null) client.stop();
        client = new IrcClient(proxy,port);
        client.start();
    }

    public static boolean isEnabled(){
        if(IRC.getINSTANCE()==null) return false;
        return IRC.getINSTANCE().enable.getValue();
    }

    public static boolean isChatSend(){
        if(IRC.getINSTANCE()==null) return false;
        return IRC.getINSTANCE().isEnabled();
    }

}

File: src/main/java/club/deneb/client/irc/IrcManager.kt
package club.deneb.client.irc

import club.deneb.client.features.ModuleManager
import club.deneb.client.features.modules.client.IRC

@Suppress("DEPRECATION")
class IrcManager {

    @JvmField
    var client: IrcClient
    private val proxy = "irc.deneb.club"
    private val port = 20730

    fun reconnect() {
        client.stop()
        client = IrcClient(proxy, port)
        client.start()
    }

    companion object {
        val isEnabled: Boolean
            get() = (ModuleManager.getModule(IRC::class.java) as IRC).enable.value
        val isChatSend: Boolean
            get() = ModuleManager.getModule(IRC::class.java).isEnabled
    }

    init {
        client = IrcClient(proxy, port)
        client.start()
    }
}